ruby/extractor/src/main.rs

```diff
@@ -168,6 +170,16 @@ fn main() -> std::io::Result<()> {
                 }
                 code_ranges = ranges;
             } else {
+                if let Some(encoding) = scan_coding_comment(&source) {
+                    is_utf8 = encoding.whatwg_name().unwrap_or_default() == "utf-8";
+                    if !is_utf8 {
+                        source = encoding
+                            .decode(&source, encoding::types::DecoderTrap::Replace)
+                            .expect("character decoding failure")
```

Comment:
`expect` will panic if `decode` returns `Err`, but we would want to just skip processing of that file, rather than aborting extraction entirely. Does the use of `DecoderTrap::Replace` guarantee that `decode` will always return `Ok`?

Reply:
It should not fail, but I rewrote the code anyway to print a warning instead.